core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.component.html

```diff
@@ -17,10 +17,10 @@
   </div>
   <div class="create-button-mobile">
     <div ng-if="($ctrl.activeTab === $ctrl.TAB_NAME_TOPICS) && $ctrl.totalTopicSummaries.length">
-      <i class="fa fa-plus-circle" aria-hidden="true" ng-click="$ctrl.createTopic()" ng-if="$ctrl.userCanCreateTopic"></i>
+      <i class="fa fa-plus-circle" aria-hidden="true" ng-click="$ctrl.createTopic()" ng-if="$ctrl.userCanCreateTopic" tabindex="-1"></i>
```

Comment:
Just to check, is it correct to assume that all `aria-hidden="true"` tags should also include `tabindex="-1"` since they need not be accessible by sequential keyboard navigation?

Reply:
Thanks! The page mentions that `Any element with a tabindex set to a numeric value other than -1` is focusable and it should not have `[aria-hidden="true"]`.

See [HTML specs](https://html.spec.whatwg.org/multipage/interaction.html#attr-tabindex) regarding the default value of tabindex:

> The default value is 0 if the element is an a, area, button, frame, iframe, input, object, select, textarea, or SVG a element, or is a summary element that is a summary for its parent details. The default value is −1 otherwise.

For `i` tag the default should be `-1` so this shouldn't have errored in the first place. Any idea why it did?

Reply:
Yea... That's weird.. I have a feeling it has something to do with ng-click attribute in each `<i> ` tag which could convert it into a `<button>` tag. This makes sense to me because screen readers announce html elements with ng-click attributes as buttons.

Reply:
See [Angular Accessibility Guide](https://docs.angularjs.org/guide/accessibility)
> If ng-click or ng-dblclick is encountered, ngAria will add tabindex="0" to any element not in the list of built in aria nodes: Button Anchor Input Textarea Select Details/Summary To fix widespread accessibility problems with ng-click on div elements, ngAria will dynamically bind a keypress event by default as long as the element isn't in a node from the list of built in aria nodes. You can turn this functionality on or off with the bindKeypress configuration option. ngAria will also add the button role to communicate to users of assistive technologies. This can be disabled with the bindRoleForClick configuration option. For ng-dblclick, you must still manually add ng-keypress and a role to non-interactive elements such as div or taco-button to enable keyboard access.

Reply:
Ah, I see. Thanks for checking!

In that case, going back to my original question, would it be reasonable to assume that all `i` tags with `aria-hidden="true"` and `ng-click` attribute needs to have a `tabindex="-1"`?

e.g.

https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-icons.component.html#L1

and

https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.html#L245

Reply:
@kevintab95 Yes I think is reasonable.

Reply:
OK thanks for confirming. Hm, in that case, can we fix such cases as well and is there a plan to prevent this from slipping through in the future (e.g. a lint check or accessibility test)?